src/utils/regex.cc

```diff
@@ -73,6 +73,11 @@ Regex::Regex(const std::string& pattern_, bool ignoreCase)
     PCRE2_SIZE erroroffset = 0;
     m_pc = pcre2_compile(pcre2_pattern, PCRE2_ZERO_TERMINATED,
         pcre2_options, &errornumber, &erroroffset, NULL);
+
+    m_pcje = pcre2_jit_compile(m_pc, PCRE2_JIT_COMPLETE);
+    m_pmc = pcre2_match_context_create(NULL);
+    m_pcjs = pcre2_jit_stack_create(32*1024, 512*1024, NULL);
```

Comment:
Just my 2c question: why does it need the own stack handling?

I do not find the article now, where the author wrote unless there is a very compelling reason, you don't need to use `pcre_jit_stack_create()`.

Few months ago I've started to play with PCRE2 JIT implementation - you can find it [here](https://github.com/digitalwave/msc_retest/tree/pcre2support-jit-without-stack). In performance, there isn't any difference between controlling own stack with huge memory like this, or without own stack. 

I also checked Nginx, it also [uses](https://github.com/nginx/nginx/blob/master/src/core/ngx_regex.c) pcre2 JIT without own stack.

Reply:
It is actually a copy-pasta from https://www.pcre.org/current/doc/html/pcre2jit.html#SEC10
I don't know if there is any penalty doing so, and I'm fine removing this.

Reply:
Tried without PCRE2 jit stack and the memory usage dropped by 50%. That's nice.

I do worry about situations like https://www.mail-archive.com/bug-grep@gnu.org/msg05763.html

It depends on the rules, though.